### DISCUSSION TITLE
Which best practices in tailwind CSS? Inline CSS or @apply CSS?

### DISCUSSION THREAD

**Discussion by parulbhayani** [tailwindlabs/tailwindcss#9459] [Help]
Which is the best practice for a long project? I will use ClassName in inline CSS or Make the CSS file and use @apply CSS?? What is the difference two types of CSS? Suggest to me the best way so I will implement it in my project.

**Comment by wongjn** [comment]
Personally, I try to avoid using `@apply` and rely on abstracting the classes as template components/partials, similar to [how the documentation describes](https://tailwindcss.com/docs/reusing-styles#extracting-components-and-partials). Abstracting as template partials means I get to stay in my template context/frame-of-mind and not have to switch to a CSS context. This means I'm not interrupted in my coding "flow". For many projects, I never have to touch my CSS file(s) at all after setup.

**Reply by adamwathan** [reply]
Yep agree with all of this, use utility classes in your HTML as much as possible 👍🏻

**Reply by parulbhayani** [reply]
If pass so many utility classes in the HTML file, then the code file is soo many classes generate that is the right way?

**Comment by noobSam1122** [comment]
If styles share between more than one component and you're not going to change is anymore, extract to @apply is a good practice , like ripple effect.
If the utility classes is related to the appearance , like bg , border,border-radius,shadow. I recommended to use in html
